scheduler/core/time.py

```diff
@@ -0,0 +1,21 @@
+from datetime import datetime, timedelta
+from math import ceil
+
+from lucupy.minimodel import NightIndex
+
+
+class TimeHandler:
+
+    def __init__(self, num_nights_to_schedule, start, time_slot_length):
```

Comment:
Type hints? Why not make this a dataclass if you're just setting members? It doesn't look like you're changing the values as far as I can tell, so it could even be:

```python
@dataclass(frozen=True)
class TimeHandler:
    ...
```